kakarot_scripts/utils/deploy_oz.py

```diff
@@ -0,0 +1,106 @@
+# %% Imports
+import asyncio
+import json
+
+import boto3
+
+from kakarot_scripts.utils.starknet import declare, deploy_starknet_account
+
+# Initialize AWS Secrets Manager client
+client = boto3.client("secretsmanager")
+
+
+# %% Store secret in AWS
+def store_secret_in_aws(secret_name, private_key):
+    """Store the private key in AWS Secrets Manager. If the secret already exists, it will be updated."""
+    try:
+        response = client.update_secret(SecretId=secret_name, SecretString=private_key)
+        secret_id = response["ARN"]
+    except client.exceptions.ResourceNotFoundException:
+        response = client.create_secret(Name=secret_name, SecretString=private_key)
+        secret_id = response["ARN"]
+    return secret_id
+
+
+# %% Save accounts to JSON
+def save_accounts_to_json(
+    accounts_data, filename="kakarot_scripts/data/oz_accounts.json"
+):
+    """Save the account data to a JSON file."""
+    with open(filename, "w") as json_file:
+        json.dump(accounts_data, json_file, indent=4)
+
+
+# %% Get private key
+def get_private_key(private_key):
+    """Retrieve or generate a private key."""
+    secret_name = "pk_oz_deploy"
+    if private_key == "":
+        try:
+            response = client.get_secret_value(SecretId=secret_name)
+            return response["SecretString"]
+        except client.exceptions.ResourceNotFoundException as err:
+            raise Exception(
+                f"Secret {secret_name} not found. Please ensure the private key is stored in AWS Secrets Manager."
+            ) from err
+    else:
+        try:
+            response = client.update_secret(
+                SecretId=secret_name, SecretString=private_key
+            )
+            _ = response["ARN"]
+            return private_key
+        except client.exceptions.ResourceNotFoundException:
+            _ = store_secret_in_aws(secret_name, private_key)
+            return private_key
+
+
+# %% Deploy accounts
+async def deploy_accounts(
+    num_accounts, amount, private_key, _class_hash, secret_name_prefix="oz_account"
+):
+    """Deploy 'num_accounts' accounts and store the private keys in AWS Secrets Manager."""
+    accounts_data = []
```

Comment:
```suggestion
def main():
    private_key = get_private_key()
    accounts_data = [
        await deploy_starknet_account(
            class_hash=class_hash, private_key=private_key, amount=amount
        )
    ]
    json.dump(accounts_data, indent=2)
```

Reply:
I kept this with the for loop because I need to call `deploy_starknet_account` a number of times equal to the number of `oz_account` that need to be generated. What do you think?

Reply:
I missed the iterator in the list comprehension, but it doesn't change anything

Reply:
fixed